Error Creating the stablecoin - For no KYC selected

**Describe the bug**
When creating a stablecoin through the Web UI, and specifically selecting not to have KYC enabled, the interface throws an error 

**To Reproduce**
Steps to reproduce the behavior:

1. Clone the repo
2. Run npm install
3. Configure env files as required
4. Run npm run start in "web" subdirectory
5. Select HashPack wallet
6. Select Testnet and click Accept
7. Create a new stablecoin - specifically select not to have KYC enabled during the process, otherwise use the default control options.
8. Submit the transaction to create the coin, and sign off the transactions in the connected wallet.
9. Observe the error in the Web UI / console (the wallet report successful transactions).
10. The stablecoin is still created (successfully?).

**Expected behavior**
No error is shown upon creating the stablecoin

**Screenshots**
<img width="1910" alt="image" src="https://github.com/hashgraph/stablecoin-studio/assets/163220202/00524af1-ab32-4cf1-9445-06f9de33c387">

**Desktop (please complete the following information):**
- OS: WSL2 (Ubuntu 22.04.3 LTS) under Windows 11 (22631.3155)
- Browser: Brave
- Version: v1.63.165 (Chromium 122.0.6261.94)

**Additional context**
- Error text shown in the console:
```
2024-03-15T01:46:14.183Z - [ERROR]	20006 - The coin 0.0.3711618 does not have KYC active	
StableCoinCreation.tsx:390 Error: The coin 0.0.3711618 does not have KYC active
    at GrantKycCommandHandler.execute (GrantKycCommandHandler.js:66:1)
    at async StableCoinInPort.grantKyc (StableCoin.js:299:1)
    at async LogError.descriptor.value (LogErrorDecorator.js:25:1)
    at async SDKService.grantKyc (SDKService.ts:492:1)
    at async handleFinish (StableCoinCreation.tsx:360:1)
```
- Create Token Transaction List: https://hashscan.io/testnet/transactionsById/0.0.3705474@1710467113.631940469
- Associate Token Transaction: https://hashscan.io/testnet/transactionsById/0.0.3705474@1710467147.425319141

--- a/web/src/views/StableCoinCreation/ManagementPermissions.tsx
+++ b/web/src/views/StableCoinCreation/ManagementPermissions.tsx
@@ -77,6 +77,17 @@ const ManagementPermissions = ({
 		}
 	}, [watch('kycKey'), watch('supplyKey')]);
 
+	useEffect(() => {
+		setValue('isKycRequired', isKycRequired);
+		if (!isKycRequired) {
+			request.kycRoleAccount = undefined;
+			request.grantKYCToOriginalSender = false;
+			setValue('kycKey', undefined);
+			setValue('kycRoleAccount', undefined);
+			setValue('grantKYCToOriginalSender', false);
+		}
+	}, [watch('kycRequired')]);
+
 	const keys = [
 		{
 			name: 'wipeKey',
@@ -220,7 +231,7 @@ const ManagementPermissions = ({
 					)}
 					{isKycRequired === true &&
 						watch('kycKey')?.value === 2 &&
-						wallet.lastWallet === SupportedWallets.HASHPACK && (
+						wallet.lastWallet !== SupportedWallets.METAMASK && (
 							<Stack minW={400}>
 								<HStack mb={4} justifyContent='space-between'>
 									<Text

--- a/web/src/views/StableCoinCreation/Review.tsx
+++ b/web/src/views/StableCoinCreation/Review.tsx
@@ -112,7 +112,7 @@ const Review = (props: ReviewProps) => {
 		roleValue: { value: number; label: string },
 		keyValue: { value: number; label: string },
 	) => {
-		if (isKycRequired || (keyValue && keyValue.value === 2)) {
+		if (isKycRequired && keyValue && keyValue.value === 2) {
 			setRoleAccountInfo(label, roleValue);
 		}
 	};

